backend/src/main/java/au/com/agiledigital/idea_search/rest/index.java

```
@@ -197,7 +187,101 @@ public String getSchemaIds( @Context HttpServletResponse response){
+    status = status != null ? status : "";
+    owner = owner != null ? owner : "";
+
+    List<FedexIdea> allIdeas = this.fedexIdeaService.queryAllFedexIdea(title, description, status, owner);
+
+    List<Map> preConvert = allIdeas.stream().map( idea -> {
+      Map preJsonIdea = new HashMap<String, String>();
+      preJsonIdea.put("title", idea.getTitle());
+      preJsonIdea.put("url", idea.getUrl());
+      preJsonIdea.put("description", idea.getDescription().isEmpty() ? "":idea.getDescription());
+      preJsonIdea.put("technologies", idea.getTechnologies().isEmpty() ? "" : idea.getTechnologies().stream().map(tech-> tech.getTechnology()).collect(
+        Collectors.toList()));
+
+      preJsonIdea.put("owner", idea.getOwner());
+
+      preJsonIdea.put("status", idea.getStatus());
+
+      return preJsonIdea;
+    }).collect(Collectors.toList());
+
+    return allIdeas.isEmpty()
+      ? "[{}]"
+      : this.gson.toJson(preConvert);
+  }
+
+  private static String extractPostRequestBody(HttpServletRequest request) throws IOException {
+    if ("POST".equalsIgnoreCase(request.getMethod())) {
+      try {
+        Scanner s = new Scanner(request.getInputStream(), "UTF-8").useDelimiter("\\A");
+        return s.hasNext() ? s.next() : "";
+      } catch (Exception e){
+        return e.toString();
+      }
+    }
+    return "";
+  }
 
+  /**
+   * @param searchString to find technologies that begin with this string
+   * @param response     Servlet contest
+   * @return String in the form of a json list of TechnologyAPI objects
+   */
+  @Path("/schema")
+  @Consumes({"application/json"})
+  @POST
+  public String postSchema(
+    @Context HttpServletRequest request,
+    @Context HttpServletResponse response
+  ) {
+    this.applyNoCacheHeaders(response);
+    String schemaBody = "";
+
+    try {
+      schemaBody = extractPostRequestBody(request);
+    } catch (Exception e){
+      throw new Error("Error parsing request body");
```

`extractPostRequestBody` already swallows all exceptions and returns the string representation of the exception (for some reason) - what is this trying to do?

---

Updated to throw error instead of returning a string representation of it. That was the original intention.

---

By doing this, you've hidden information about the original exception. Consider something like:
```
throw new Exception("Failed to parse request body.", e)
```

---

Updated!

---

Updated again to avoid rethrowing.